test_conformance/extensions/cl_khr_command_buffer/command_buffer_test_copy.cpp

```diff
@@ -0,0 +1,459 @@
+//
+// Copyright (c) 2022 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#include "basic_command_buffer.h"
+#include "harness/typeWrappers.h"
+#include "procs.h"
+
+#include <vector>
+
+
+namespace {
+
+////////////////////////////////////////////////////////////////////////////////
+// Command-queue fill tests which handles below cases:
+// 
+// -copy image
+// -copy buffer
+// -copy buffer to image
+// -copy image to buffer
+// -copy buffer rect
+
+    struct CopyImageKHR : public BasicCommandBufferTest
+    {
+        using BasicCommandBufferTest::BasicCommandBufferTest;
+
+        cl_int Run() override
+        {
+            cl_int error = clCommandFillImageKHR(
+                command_buffer,
+                nullptr,
+                src_image,
+                fill_color,
+                origin,
+                region,
+                0,
+                nullptr,
+                nullptr,
+                nullptr);
+
+            test_error(error, "clCommandFillImageKHR failed");
+
+            error = clCommandCopyImageKHR(
+                    command_buffer,
+                    nullptr,
+                    src_image,
+                    dst_image,
+                    origin,
+                    origin,
+                    region,
+                    0,
+                    0,
+                    nullptr,
+                    nullptr);
+
+            test_error(error, "clCommandCopyImageKHR failed");
+
+            error = clFinalizeCommandBufferKHR(command_buffer);
+            test_error(error, "clFinalizeCommandBufferKHR failed");
+
+            error = clEnqueueCommandBufferKHR(0, nullptr, command_buffer, 0,
+                                            nullptr, nullptr);
+            test_error(error, "clEnqueueCommandBufferKHR failed");
+
+            error = clFinish(queue);
+
+            test_error(error, "clFinish failed");
+
+            std::vector<cl_char> output_data(data_size);
+            error = clEnqueueReadImage(queue, dst_image, CL_TRUE, origin, region, 0, 0, output_data.data(), 0, nullptr, nullptr);
```

Comment:
The same comment as for `command_buffer_test_fill.cpp`: Normally `clFinish` should be called after `clEnqueueReadBuffer` but in this case it is not necessary at all due to blocking read operation (third parameter set to CL_TRUE)

Reply:
Done.